File: p169.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> counts;
        for(int num : nums) {
            counts[num] += 1;
        }
        int max = INT_MIN;
        int value = 0;
        for(auto key_value : counts) {
            if(key_value.second > max) {
                max = key_value.second;
                value = key_value.first;
            }
        }
        return value;
    }
};

File: p169.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        counts = {}
        for i in nums:
            if i not in counts:
                counts[i] = 0
            counts[i] += 1
        max_count = 0
        solution = None
        for i in counts:
            if counts[i] > max_count:
                max_count = counts[i]
                solution = i
        return solution